projects/jobs/jobs.groovy

```diff
@@ -211,17 +268,77 @@ xmlFiles.each {
     configXml.getBytes());
 
   String jobName = configFile.getName()
-  		.substring(0,
+      .substring(0,
                    configFile
                    .getName()
-                   	.lastIndexOf('.'));
+                    .lastIndexOf('.'));
 
   jenkinsInstace.getItem(projectName,jenkinsInstace)
-  	.createProjectFromXML(jobName, xmlStream);
+    .createProjectFromXML(jobName, xmlStream);
 
   println '[INFO] - Imported XML job config: ' + it.toURI();
 }
 ''')
+  environmentVariables {
+      env('PLUGGABLE_SCM_PROVIDER_PATH','${WORKSPACE}/job_dsl_additional_classpath/')
+      env('PLUGGABLE_SCM_PROVIDER_PROPERTIES_PATH','${WORKSPACE}/datastore/pluggable/scm')
+  }
+  groovy {
+    scriptSource {
+        stringScriptSource {
+            command('''import pluggable.scm.SCMProvider;
+import pluggable.scm.SCMProviderHandler;
+import pluggable.configuration.EnvVarProperty;
+
+EnvVarProperty envVarProperty = EnvVarProperty.getInstance();
+envVarProperty.setVariableBindings(System.getenv());
+
+String scmProviderId = envVarProperty.getProperty('SCM_PROVIDER_ID')
+
+SCMProvider scmProvider = SCMProviderHandler.getScmProvider(scmProviderId, System.getenv())
+
+def workspace = envVarProperty.getProperty('WORKSPACE')
+def projectFolderName = envVarProperty.getProperty('PROJECT_NAME')
+def cartridgeFolder = envVarProperty.getProperty('CARTRIDGE_FOLDER')
```

Comment:
If CARTRIDGE_FOLDER is not provided we will get a missing property exception. I propose we do something similar to the SCM_NAME space below and default it to '' if it's not provided.